mcci_catena_provision_chirpstack.py

```diff
@@ -0,0 +1,1280 @@
+        return True
+    else:
+        oAppContext.error('Port {} already closed'.format(sPortName))
+        return None
+
+##############################################################################
+#
+#   main 
+#
+##############################################################################
+
+if __name__ == '__main__':
+        
+    pName = os.path.basename(__file__)
+    pDir = os.path.dirname(os.path.abspath(__file__))
+    
+    deviceTypeUrl = ''
+    deviceUrl = ''
+    orgsUrl = ''
+    appsUrl = ''
+    devProfileUrl = ''
+    registerDeviceUrl = ''
+    getDeviceInfoUrl = ''
+
+    oAppContext = AppContext()
+
+    optparser = argparse.ArgumentParser(
+        description='MCCI Catena Provisioning')
+    optparser.add_argument(
+        '-baud',
+        action='store',
+        nargs='?',
+        dest='baudrate',
+        type=int,
+        help='Specify the baud rate as a number. Default is 115200')
+    optparser.add_argument(
+        '-port',
+        action='store',
+        nargs=1,
+        dest='portname',
+        type=str,
+        required=True,
+        help='Specify the COM port name. This is system specific')
+    optparser.add_argument(
+        '-D',
+        action='store_true',
+        default=False,
+        dest='debug',
+        help='Operate in debug mode. Causes more output to be produced')
+    optparser.add_argument(
+        '-info',
+        action='store_true',
+        default=False,
+        dest='info',
+        help='Display the Catena info')
+    optparser.add_argument(
+        '-v',
+        action='store_true',
+        default=False,
+        dest='verbose',
+        help='Operate in verbose mode')
+    optparser.add_argument(
+        '-echo',
+        action='store_true',
+        default=False,
+        dest='echo',
+        help='Echo all device operations')
+    optparser.add_argument(
+        '-V',
+        action='append',
+        dest='vars',
+        help='Specify ttn config info in name=value format')
+    optparser.add_argument(
+        '-nowrite',
+        action='store_false',
+        default=True,
+        dest='writeEnable',
+        help='Disable writes to the device')
+    optparser.add_argument(
+        '-permissive',
+        action='store_true',
+        default=False,
+        dest='permissive',
+        help='Don\'t give up if SYSEUI isn\'t set.')
+    optparser.add_argument(
+        '-r',
+        action='store_true',
+        default=False,
+        dest='register',
+        help='Register the device in actility network')
+    optparser.add_argument(
+        '-Werror',
+        action='store_true',
+        default=False,
+        dest='warning',
+        help='Warning messages become error messages')
+    optparser.add_argument(
+        '-s',
+        action='store',
+        nargs=1,
+        dest='script',
+        type=str,
+        help='Specify script name to load catena info')
+
+    opt = optparser.parse_args()
```

Comment:
Please move this to a separate function, called at this point.  I usually call it `ParseArguments()`. The function should return the result of `optparser.parse_args()`.  This is for clarity.

Reply:
Moved the argparser code to the separate function `parse_arguments()`